packages/connection-manager/src/js/Connection/XHRConnection.js

```diff
@@ -0,0 +1,229 @@
+import AbstractConnection from "./AbstractConnection";
+import ConnectionEvent from "../ConnectionEvent/ConnectionEvent";
+
+export const ALLOWED_METHODS = ["GET", "POST", "PUT", "DELETE"];
+
+/**
+ * Basic XHR Connection
+ * @todo add/remove headers
+ */
+export default class XHRConnection extends AbstractConnection {
+  /**
+   * Initializes an Instance of XHRConnection
+   * @constructor
+   * @param {string} url – URL to be fetched
+   * @param {object} [options]
+   * @param {string} [options.method=GET] – used method
+   * @param {*} [options.body] – payload for request
+   * @param {string} [options.headers] – additional headers (like content-type)
+   */
+  constructor(url, options = {}) {
+    super(url);
+
+    const {
+      method = "GET",
+      body = null,
+      headers = null,
+      responseType = "json"
+    } = options;
+
+    let _method;
+    Object.defineProperty(this, "method", {
+      get: () => _method,
+      set: __method => {
+        if (ALLOWED_METHODS.includes(__method) === -1) {
+          throw new Error("Invalid method for XHR Request.");
+        }
+        _method = __method;
```

Comment:
Not blocking, but do really need all these underscores?  Let's please not use a hungarian notation to flag them as private.